packages/typespec-ts/test/integration/azureClientGeneratorCoreUsage.spec.ts

```diff
@@ -37,4 +37,16 @@ describe("Usage Client", () => {
       assert.fail(err as string);
     }
   });
+
+  it("should put usage model in operation", async () => {
+    try {
+      const result = await client
+        .path("/azure/client-generator-core/usage/modelInReadOnlyProperty")
+        .put({ body: {} });
+      assert.strictEqual(result.status, "200");
+      assert.strictEqual((result.body as any).name, "Madge");
```

Comment:
why cast as any here?

Reply:
Property `name` does not exist on type `RoundTripModelOutput`.
https://github.com/Azure/autorest.typescript/blob/f455956110b1f4331c5e84f3c01e12695327ff67/packages/typespec-ts/test/integration/generated/azure/client-generator-core/usage/src/outputModels.ts#L9

Reply:
Sounds like a cadl ranch bug?

Reply:
confirming https://github.com/Azure/cadl-ranch/pull/587/files#r1680820057